bug(core): LLM context overflow silently swallowed — chat returns 'Completed task' instead of error

## Summary
When the system prompt exceeds the LLM's actual context window, the non-streaming chat API silently returns `{reply: 'Completed task: <input>', thought: 'Completed task: <input>'}` instead of surfacing the LLM error. The error is only visible in LM Studio's logs.

## Evidence
LM Studio error log:
```
[ERROR] [qwen3.5-35b-a3b] The number of tokens to keep from the initial prompt is greater than the context length (n_keep: 5213 >= n_ctx: 4096)
```

But the API returns 200 with:
```json
{"reply": "Completed task: Use the file-list tool...", "thought": "Completed task: Use the file-list tool..."}
```

## Root Cause
The in-process BaseAgent chat path catches the LLM error and returns a fallback response ('Completed task: ...') without propagating the error. The SERA MCP server and API both see this as a successful response.

## Expected
1. The error should be surfaced to the caller — either as an error response or with clear error content in the reply
2. The ContextManager should validate that the assembled prompt fits within the model's context window BEFORE sending to the LLM
3. The providers.json contextWindow (131072) should be validated against the actual model capability

## Impact
- All chat via MCP and non-streaming API appears to work but returns meaningless responses
- Users see 'Completed task' and think the agent processed their request
- No indication that the model can't handle the prompt size

--- a/core/src/agents/BaseAgent.ts
+++ b/core/src/agents/BaseAgent.ts
@@ -430,62 +430,84 @@ export abstract class BaseAgent {
     let accumulated = '';
     let accumulatedReasoning = '';
 
-    for await (const chunk of this.llmProvider.chatStream(messages)) {
-      if (chunk.token) {
-        accumulated += chunk.token;
+    try {
+      for await (const chunk of this.llmProvider.chatStream(messages)) {
+        if (chunk.token) {
+          accumulated += chunk.token;
 
-        // First content token: flush any accumulated reasoning as one thought block
-        if (accumulatedReasoning) {
-          await this.publishThought('reasoning', accumulatedReasoning);
-          accumulatedReasoning = '';
+          // First content token: flush any accumulated reasoning as one thought block
+          if (accumulatedReasoning) {
+            await this.publishThought('reasoning', accumulatedReasoning);
+            accumulatedReasoning = '';
+          }
         }
-      }
 
-      // Accumulate reasoning/thinking tokens (e.g. Qwen / DeepSeek reasoning_content)
-      if (chunk.reasoning) {
-        accumulatedReasoning += chunk.reasoning;
-      }
+        // Accumulate reasoning/thinking tokens (e.g. Qwen / DeepSeek reasoning_content)
+        if (chunk.reasoning) {
+          accumulatedReasoning += chunk.reasoning;
+        }
 
-      if (chunk.usage && this.meteringEngine) {
-        await this.meteringEngine.record({
-          agentId: this.agentInstanceId || this.role,
-          model: this.manifest.model.name,
-          ...chunk.usage,
-        });
-      }
+        if (chunk.usage && this.meteringEngine) {
+          await this.meteringEngine.record({
+            agentId: this.agentInstanceId || this.role,
+            model: this.manifest.model.name,
+            ...chunk.usage,
+          });
+        }
 
-      // Only publish to the frontend when there is an actual token or the stream
-      // is done. Skipping empty-string tokens avoids flooding Centrifugo with
-      // hundreds of no-op HTTP requests during a thinking/reasoning phase
-      // (e.g. Qwen3 emits reasoning_content before any visible content).
-      if (this.intercom && (chunk.token !== '' || chunk.done)) {
-        await this.intercom.publishToken(
-          this.agentInstanceId || this.role,
-          chunk.token,
-          chunk.done,
-          messageId
-        );
-      }
-    }
-
-    // Flush any remaining reasoning (e.g. model only reasoned, produced no content)
-    if (accumulatedReasoning) {
-      await this.publishThought('reasoning', accumulatedReasoning);
-
-      // Fallback: if the model only produced reasoning tokens and no content tokens
-      // (happens when reasoning flag is misconfigured), use the reasoning as the reply.
-      if (!accumulated) {
-        accumulated = accumulatedReasoning;
-        // Also publish the reasoning as token content so the web UI shows it
-        if (this.intercom) {
+        // Only publish to the frontend when there is an actual token or the stream
+        // is done. Skipping empty-string tokens avoids flooding Centrifugo with
+        // hundreds of no-op HTTP requests during a thinking/reasoning phase
+        // (e.g. Qwen3 emits reasoning_content before any visible content).
+        if (this.intercom && (chunk.token !== '' || chunk.done)) {
           await this.intercom.publishToken(
             this.agentInstanceId || this.role,
-            accumulated,
-            false,
+            chunk.token,
+            chunk.done,
             messageId
           );
         }
       }
+
+      // Flush any remaining reasoning (e.g. model only reasoned, produced no content)
+      if (accumulatedReasoning) {
+        await this.publishThought('reasoning', accumulatedReasoning);
+
+        // Fallback: if the model only produced reasoning tokens and no content tokens
+        // (happens when reasoning flag is misconfigured), use the reasoning as the reply.
+        if (!accumulated) {
+          accumulated = accumulatedReasoning;
+          // Also publish the reasoning as token content so the web UI shows it
+          if (this.intercom) {
+            await this.intercom.publishToken(
+              this.agentInstanceId || this.role,
+              accumulated,
+              false,
+              messageId
+            );
+          }
+        }
+      }
+    } catch (streamError) {
+      const errMsg = streamError instanceof Error ? streamError.message : String(streamError);
+      this.logger.error(`[${this.role}] LLM stream error:`, errMsg);
+      const userMsg =
+        '⚠️ LLM error: ' +
+        (errMsg.includes('context') || errMsg.includes('n_keep') || errMsg.includes('n_ctx')
+          ? 'Context window exceeded. Try a shorter message or clear the conversation.'
+          : errMsg.length > 200
+            ? errMsg.substring(0, 200) + '...'
+            : errMsg);
+      await this.publishThought('reflect', userMsg);
+      if (this.intercom) {
+        await this.intercom.publishToken(
+          this.agentInstanceId || this.role,
+          userMsg,
+          true,
+          messageId
+        );
+      }
+      return { thought: 'LLM error', finalAnswer: userMsg };
     }
 
     const reply = accumulated || 'No response generated.';

--- a/core/src/agents/WorkerAgent.ts
+++ b/core/src/agents/WorkerAgent.ts
@@ -112,6 +112,18 @@ export class WorkerAgent extends BaseAgent {
       }
 
       // No tool calls — we have a final text response
+      // Detect empty/error responses from the LLM
+      if (!response.content && !response.toolCalls?.length) {
+        const errorMsg =
+          '⚠️ The LLM returned an empty response. This may indicate a context window overflow, ' +
+          'model misconfiguration, or provider error. Check agent and LLM provider logs for details.';
+        this.logger.warn(
+          `[${this.name}] LLM returned empty content for input: ${input.substring(0, 100)}`
+        );
+        await this.publishThought('reflect', errorMsg);
+        return { thought: 'LLM returned empty response', finalAnswer: errorMsg };
+      }
+
       this.history = [
         ...fullHistory,
         { role: 'assistant', content: response.content } as ChatMessage,

